### PR TITLE
Handle bad list item children gracefully

### PR DESCRIPTION
**Pull request by zurfyx**
Generics are a lie https://github.com/facebook/lexical/issues/3152#issuecomment-1273576531

Turns out there is a way to make this fail.. we need to investigate the root of the problem. Fixing the existing crash scenario

---

See https://github.com/facebook/lexical/issues/3227

### CODE REVIEW

File: packages/lexical-list/src/formatList.ts [hunk 1]
```diff
@@ -270,16 +270,26 @@ export function removeList(editor: LexicalEditor): void {
...
+        const prevValue = child.getValue();
+        const nextValue = $getListItemValue(child);
+
+        if (prevValue !== nextValue) {
+          child.setValue(nextValue);
+        }
+      } else {
+        if (__DEV__) {
+          invariant(false, 'Expected ListNode child to be a ListItemNode');
+        }
```
Comment (zurfyx) [inline packages/lexical-list/src/formatList.ts:289]: @fantactuka I wonder what you think about this. While I'm all in into enforcing document consistency, at this point it's already too late. The harm has already been done, the best we can do is to handle it gracefully.

On the contrary, when caught right on the update operation that's causing this the editor can revert the update before any commit has triggered

**Comment by zurfyx** [comment]
OK, I think this is just a "fix" to the actual problem but we should prioritize this https://github.com/facebook/lexical/pull/3226 which didn't stop this from happening, as well of the actual problem which is still unkown.

File: packages/lexical-list/src/formatList.ts (hunk 1 above)
Comment (acywatson) [reply packages/lexical-list/src/formatList.ts:289]: We need to change how list items are nested and we can get rid of all of this anyway

Comment (zurfyx) [reply packages/lexical-list/src/formatList.ts]: Deleted invariant

**Review by fantactuka — APPROVED** [review]

**Pull request merged**